packages/driver/src/cypress/cy.ts

```diff
@@ -504,20 +504,38 @@ export class $Cy implements ITimeouts, IStability, IAssertions, IRetries, IJQuer
         // about:blank in a visit, we do need these
         this.contentWindowListeners(autWindow)
 
-        this.Cypress.action('app:window:load', this.state('window'))
-
-        // we are now stable again which is purposefully
-        // the last event we call here, to give our event
-        // listeners time to be invoked prior to moving on
-        this.isStable(true, 'load')
+        try {
+          this.Cypress.action('app:window:load', this.state('window'))
+        } catch (err) {
+          err.isFromWindowLoadEvent = true
+
+          throw err
```

Comment:
What instance would an error be thrown here? Cypress.action will emit two events wit this change, but neither appears to be throwing an error.

Reply:
If this try-catch should is needed, should we move to stable if an error is thrown? & do we need to wrap the error to display nicely in the Console Log with help text & a stack trace?

Reply:
> What instance would an error be thrown here? Cypress.action will emit two events wit this change, but neither appears to be throwing an error.

This triggers the event `window:load` to users, so their code could error. 

```js
cy.on('window:load', () => {
 throw new Error('foo')
})
```

This needs to be differentiated from cross-origin errors below.

Any time you see `Cypress.action()`, it will usually result in an [event that users can listen to with `cy.on` or `Cypress.on`](https://docs.cypress.io/api/events/catalog-of-events#Event-Types). Internally, we prefix the event with its origin (`app:`, `runner:`, etc), then strip that when we [send it to the public API](https://github.com/cypress-io/cypress/blob/797002f0ff95b04bd8b8b1096b5bbb39a431dbc0/packages/driver/src/cypress.ts#L279). 

> If this try-catch should is needed, should we move to stable if an error is thrown?

Good question. In the case of an event handler causing the error, yes, because it's not the page load that errored. But it probably shouldn't move to stable if it's a cross-origin error, so I think I need to revise this.

> do we need to wrap the error to display nicely in the Console Log with help text & a stack trace?

The stack trace is fine as-is, but the error does need to be wrapped in our "this came from your test code" message, so I'll update that.

Reply:
> Good question. In the case of an event handler causing the error, yes, because it's not the page load that errored. But it probably shouldn't move to stable if it's a cross-origin error, so I think I need to revise this.

Addressed in [`c9c963d`](https://github.com/cypress-io/cypress/pull/19323/commits/c9c963d3e83b0b25f9f09457825ec5e632d5391e).

> The stack trace is fine as-is, but the error does need to be wrapped in our "this came from your test code" message, so I'll update that.

Addressed in [`e2d7113`](https://github.com/cypress-io/cypress/pull/19323/commits/e2d7113a5d217235703dcdb414f0bf1d53e7c8af).

Looks like this:

![Screen Shot 2022-01-03 at 10 04 02 AM](https://user-images.githubusercontent.com/1157043/147947418-ad436395-e753-47b4-a751-2082778a23fc.png)

Reply:
Actually, I realized this error shouldn't be wrapped, so I reverted that change. We only wrap errors in the 'from your test code' bit when it's uncaught.